### PR TITLE
Updating MobileNetwork SDK with new API release 2024-02-01 

### PR DESCRIPTION
**Pull request by khattartarun**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by jsquire — DISMISSED** [review]
@khattartarun.   Your account requires write permissions to the Azure SDK repository, which it does not currently have.  You'll need to correct this before this PR can move forward.

Please review the Azure SDK onboarding documentation and use the Teams channel for questions/support.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

**Comment by khattartarun** [comment]
@jsquire  Can you please review this PR?

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/HomeNetworkPublicKey.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="HomeNetworkPublicKey"/> for deserialization. </summary>
+        internal HomeNetworkPublicKey()
+        {
+        }
+
+        /// <summary> The Home Network Public Key Identifier determines which public key was used to generate the SUCI sent to the AMF. See TS 23.003 Section 2.2B Section 5. </summary>
+        public int Id { get; set; }
```
Comment (archerzz) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/HomeNetworkPublicKey.cs:72]: What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            RatType = ratType;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="UeInfo4G"/> for deserialization. </summary>
+        internal UeInfo4G()
+        {
+        }
+
+        /// <summary> International mobile subscriber identifier. </summary>
+        public string Imsi { get; set; }
```
Comment (archerzz) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:127]: From the description, looks like this property should be an identifier. Is `Imsi` a proper noun？

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        /// <summary> International mobile equipment identity. </summary>
+        public string Imei { get; set; }
+        /// <summary> International mobile equipment identity – software version. </summary>
+        public string Imeisv { get; set; }
+        /// <summary> Gets the session info. </summary>
+        public IList<UeSessionInfo4G> SessionInfo { get; }
+        /// <summary> Per-UE transport network layer association. </summary>
+        public string PerUeTnla { get; set; }
+        /// <summary> MME S1AP identifier. </summary>
+        public int? MmeS1ApId { get; set; }
```
Comment (archerzz) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:137]: Just want to double check the format of the ID

Comment (archerzz) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:137]: same question for other ID properties in this class

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            RatType = ratType;
+        }
+
+        /// <summary> Initializes a new instance of <see cref="UeInfo5G"/> for deserialization. </summary>
+        internal UeInfo5G()
+        {
+        }
+
+        /// <summary> Subscription Permanent Identifier. </summary>
+        public string Supi { get; set; }
```
Comment (archerzz) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs:132]: same question for the naming

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+        /// <summary> Permanent Equipment Identifier. </summary>
+        public string Pei { get; set; }
+        /// <summary> Gets the session info. </summary>
+        public IList<UeSessionInfo5G> SessionInfo { get; }
+        /// <summary> Per-UE transport network layer association. </summary>
+        public string PerUeTnla { get; set; }
+        /// <summary> The AMF UE NGAP ID. </summary>
+        public long? AmfUeNgapId { get; set; }
```
Comment (archerzz) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs:140]: same question for the format of the ID

**Review by archerzz — COMMENTED** [review]
Generally, two questions:
- A lot of ID properties having name ending with `i` instead of `Id`. Are those proper nouns? Otherwise we should use `Id` as the suffix.
- Just want to double confirm the formats of some ID properties. We should use `ResourceIdentifier` and `Guid` as much as possible.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/HomeNetworkPublicKey.cs (hunk 1 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/HomeNetworkPublicKey.cs:72]: This "Id" is not for resource but should be an integer value that identifies the public key. It has value 0-255. This is as per telecom specification section 2.2B https://www.[etsi.org/deliver/etsi_TS/123000_123099/123003/16.06.00_60/ts_123003v160600p.pdf](https://www.etsi.org/deliver/etsi_TS/123000_123099/123003/16.06.00_60/ts_123003v160600p.pdf)

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs (hunk 2 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:127]: Yes. As mentioned in summary it stands for International mobile subscriber identifier and widely used as "Imsi" only.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs (hunk 3 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:137]: The format should be  uint. This generated using "dotnet build /t:GenerateCode".

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs (hunk 4 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs:132]: Name is correct. Its as per telecom specification Sect 2.2A https://www.etsi.org/deliver/etsi_TS/123000_123099/123003/16.06.00_60/ts_123003v160600p.pdf

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs (hunk 5 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo5G.cs:140]: Yes. It must be long. But I am not sure why  "dotnet build /t:GenerateCode" generated it as nullable long.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs (hunk 3 above)
Comment (khattartarun) [reply sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/Generated/Models/UeInfo4G.cs:137]: Refer https://github.com/Azure/azure-rest-api-specs/blob/main/specification/mobilenetwork/resource-manager/Microsoft.MobileNetwork/stable/2024-02-01/ueInfo.json

**Comment by khattartarun** [comment]
Hi @archerzz I have replied to your comments. Naming used for IDs are as per specification. The data types are generated using "dotnet build /t:GenerateCode". Do let me know if I need to change something.

**Comment by khattartarun** [comment]
> Generally, two questions:
> 
> * A lot of ID properties having name ending with `i` instead of `Id`. Are those proper nouns? Otherwise we should use `Id` as the suffix.
> * Just want to double confirm the formats of some ID properties. We should use `ResourceIdentifier` and `Guid` as much as possible.

Yes names are correct and as per speificiation we follow for 5G technology. All these IDs are available in swagger https://github.com/Azure/azure-rest-api-specs/blob/main/specification/mobilenetwork/resource-manager/Microsoft.MobileNetwork/stable/2024-02-01/ueInfo.json and https://github.com/Azure/azure-rest-api-specs/blob/main/specification/mobilenetwork/resource-manager/Microsoft.MobileNetwork/stable/2024-02-01/ts29571.json

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
@khattartarun Can you fix the conflict? Besides, 
if you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

**Comment by khattartarun** [comment]
> @khattartarun Can you fix the conflict? Besides, if you are going to release a new version, don't forget to update `CHANGELOG.md`. See [here](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=signing-and-publishing-nuget-packges) for details.

Resolved the conflict now. 

Updated the change log by following this guideline - https://eng.ms/docs/products/azure-developer-experience/develop/sdk-release-mgmt-dotnet. Please check.

File: .vscode/cspell.json [hunk 6]
```diff
@@ -623,7 +623,29 @@
...
+        "tngf",
+        "fiveqi",
+        "plmn",
+        "napt",
+        "nssai",
+        "iccid",
+        "gbps",
+        "mbps",
+        "sdkutrg",
+        "ue's"
```
Comment (ArthurMa1978) [inline .vscode/cspell.json:648]: is this right?

Comment (khattartarun) [reply .vscode/cspell.json:648]: yes all these are right keywords

**Review by jsquire — DISMISSED** [review]
Why are there Azure Monitor changes in this PR?   Please rebase and ensure only the files that you're intending to change are included.

**Comment by khattartarun** [comment]
> Why are there Azure Monitor changes in this PR? Please rebase and ensure only the files that you're intending to change are included.

Sure. Got this added by running a script. Let me remove them.

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**